### PR TITLE
[fix](jdbc catalog) fix npe when use only_specified_database config and jdbc_url has schema

### PR DESCRIPTION
**Pull request by imbls**
## Proposed changes

Issue Number: close #xxx

set dbname mapping when use only_specified_database config and jdbc_url has schema

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by imbls** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/jdbc/client/JdbcClient.java [hunk 1]
```diff
@@ -363,7 +363,7 @@ protected List<String> getSpecifiedDatabase(Connection conn) {
         } finally {
             close(conn);
         }
-        return databaseNames;
+        return jdbcLowerCaseMetaMatching.setDatabaseNameMapping(databaseNames);
```
Comment (zy-kkk) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/jdbc/client/JdbcClient.java:366]: protected List<String> getSpecifiedDatabase(Connection conn) {
        List<String> remoteDatabaseNames = Lists.newArrayList();
        try {
            remoteDatabaseNames.add(conn.getSchema());
        } catch (SQLException e) {
            throw new JdbcClientException("failed to get specified database name from jdbc", e);
        } finally {
            close(conn);
        }
        return filterDatabaseNames(remoteDatabaseNames);
    }

File: fe/fe-core/src/main/java/org/apache/doris/datasource/jdbc/client/JdbcMySQLClient.java [hunk 2]
```diff
@@ -77,7 +77,7 @@ protected List<String> getSpecifiedDatabase(Connection conn) {
         } finally {
             close(conn);
         }
-        return databaseNames;
+        return jdbcLowerCaseMetaMatching.setDatabaseNameMapping(databaseNames);
```
Comment (zy-kkk) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/jdbc/client/JdbcMySQLClient.java:80]: @Override
    protected List<String> getSpecifiedDatabase(Connection conn) {
        List<String> remoteDatabaseNames = Lists.newArrayList();
        try {
            remoteDatabaseNames.add(conn.getCatalog());
        } catch (SQLException e) {
            throw new JdbcClientException("failed to get specified database name from jdbc", e);
        } finally {
            close(conn);
        }
        return filterDatabaseNames(remoteDatabaseNames);
    }

File: fe/fe-core/src/main/java/org/apache/doris/datasource/jdbc/client/JdbcClient.java [hunk 3]
```diff
@@ -221,7 +221,7 @@ public List<String> getDatabaseNameList() {
         Statement stmt = null;
         ResultSet rs = null;
         if (isOnlySpecifiedDatabase && includeDatabaseMap.isEmpty() && excludeDatabaseMap.isEmpty()) {
-            return getSpecifiedDatabase(conn);
+            return filterDatabaseNames(getSpecifiedDatabase(conn));
```
Comment (zy-kkk) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/jdbc/client/JdbcClient.java:224]: You should also apply this modification where you override getDatabaseNameList()

Comment (imbls) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/jdbc/client/JdbcClient.java:224]: ok, it’s my mistake, i will modify in tonight

**Pull request closed without merging**